pkg/controller/fileshare/filesharecontroller_test.go

```diff
@@ -0,0 +1,217 @@
+	"google.golang.org/grpc"
+)
+
+type fakefileshareClient struct{}
+
+func NewFakeClient() client.Client {
+	return &fakefileshareClient{}
+}
+
+func (fc *fakefileshareClient) Connect(edp string) error {
+	return nil
+}
+
+func (fc *fakefileshareClient) Close() {
+	return
+}
+
+func (fc *fakefileshareClient) CreateFileShare(ctx context.Context, in *pb.CreateFileShareOpts, opts ...grpc.CallOption) (*pb.GenericResponse, error) {
+	return &pb.GenericResponse{
+		Reply: &pb.GenericResponse_Result_{
+			Result: &pb.GenericResponse_Result{
+				Message: ByteFileShare,
+			},
+		},
+	}, nil
+}
+
+func (fc *fakefileshareClient) CreateFileShareAcl(ctx context.Context, in *pb.CreateFileShareAclOpts, opts ...grpc.CallOption) (*pb.GenericResponse, error) {
+	return &pb.GenericResponse{
+		Reply: &pb.GenericResponse_Result_{
+			Result: &pb.GenericResponse_Result{
+				Message: ByteFileShareAcl,
+			},
+		},
+	}, nil
+}
+
+// DeleteFileShare provides a mock function with given fields: ctx, in, opts
+func (fc *fakefileshareClient) DeleteFileShare(ctx context.Context, in *pb.DeleteFileShareOpts, opts ...grpc.CallOption) (*pb.GenericResponse, error) {
+	return &pb.GenericResponse{
+		Reply: &pb.GenericResponse_Result_{
+			Result: &pb.GenericResponse_Result{
+				Message: ByteFileShare,
+			},
+		},
+	}, nil
+}
+
+func (fc *fakefileshareClient) CreateFileShareSnapshot(ctx context.Context, in *pb.CreateFileShareSnapshotOpts, opts ...grpc.CallOption) (*pb.GenericResponse, error) {
+	return &pb.GenericResponse{
+		Reply: &pb.GenericResponse_Result_{
+			Result: &pb.GenericResponse_Result{
+				Message: ByteFileShareSnapshot,
+			},
+		},
+	}, nil
+}
+
+func (fc *fakefileshareClient) DeleteFileShareSnapshot(ctx context.Context, in *pb.DeleteFileShareSnapshotOpts, opts ...grpc.CallOption) (*pb.GenericResponse, error) {
+	return &pb.GenericResponse{
+		Reply: &pb.GenericResponse_Result_{
+			Result: &pb.GenericResponse_Result{
+				Message: ByteFileShareSnapshot,
+			},
+		},
+	}, nil
+}
+
+func NewFakeController() Controller {
+	return &controller{
+		Client:   NewFakeClient(),
+		DockInfo: &model.DockSpec{},
+	}
+}
+
+func TestCreateFileShare(t *testing.T) {
+	fc := NewFakeController()
+	var expected = &SampleShares[0]
+
+	result, err := fc.CreateFileShare(&pb.CreateFileShareOpts{})
+	if err != nil {
+		t.Errorf("Failed to create fileshare, err is %v\n", err)
+	}
+
+	if !reflect.DeepEqual(result, expected) {
+		t.Errorf("Expected %v, got %v\n", expected, result)
+	}
+}
+
+func TestDeleteFileShare(t *testing.T) {
+	fc := NewFakeController()
+
+	result := fc.DeleteFileShare(&pb.DeleteFileShareOpts{})
+	if result != nil {
+		t.Errorf("Expected %v, got %v\n", nil, result)
+	}
+}
+
+func TestCreateFileShareSnapshot(t *testing.T) {
+	fc := NewFakeController()
+	var expected = &SampleShareSnapshots[0]
+
+	result, err := fc.CreateFileShareSnapshot(&pb.CreateFileShareSnapshotOpts{})
+	if err != nil {
+		t.Errorf("Failed to create file share snapshot, err is %v\n", err)
```

Comment:
The first letter of error msg should be lower case

Reply:
Done